source/common/modules/markdown-editor/hooks/note-preview.js

```diff
@@ -0,0 +1,100 @@
+const tippy = require('tippy.js').default
+const { ipcRenderer } = require('electron')
+
+/**
+ * A hook for displaying link tooltips which can be used to visually
+ * enable users to click a link (without having to press down Ctrl/Cmd)
+ *
+ * @param   {CodeMirror}  cm  The instance to attach to
+ */
+module.exports = (cm) => {
+  cm.getWrapperElement().addEventListener('mousemove', (event) => {
+    let a = event.target
+
+    // Only for note links
+    if (!a.classList.contains('cm-zkn-link')) {
+      return
+    }
+
+    // If there's already a tippy on the instance, don't re-render it
+    if (a.hasOwnProperty('_tippy')) {
+      return
+    }
+
+    // Initialise displayed attributes to 'loading'
+    let title = 'loading'
+    let content = 'loading'
```

Comment:
btw wouldn't it make sense to add an ellipsis here to indicate a process …? Or, even better, a loading spinner?

Reply:
Would you have any tips on where to put the gif for this spinner?

Reply:
I don't know if we should use a GIF – we could do this readily with SVG or even JavaScript-animated HTML. After all, we do have the space, and a single, continuous animation shouldn't be too demanding (but the quality improvements are massive).

One quick Google search yielded, e.g., this one: https://loading.io/css/

And I remember I had a loading spinner animation in an older version of Zettlr, maybe we could recycle that one by having a look into the old versions, it should be ~somewhere~ there!

Reply:
Sure, I'll put one of those in. Would you be able to tell me where to put the css for the note-preview.js file I made?

Reply:
hm... "Somewhere reasonable", I don't have the source code in front of me, so I'll leave the decision to you for now. I mean moving around a file shouldn't be a deal breaker! :)

Reply:
I am a bit stuck with trying to get my css into my javascript. From my research, I can simply import the document. However this file must use the 'require' method, and even then I cannot find the file.

My chosen location for the file is a in a folder called css in the hooks folder. Therefore, the following code should work:

const { tippyCSS } = require('css/note-preview.css')

right? Let me know if I am way off here, im not quite sure with this stuff.